tools/analyzer_plugin/README.md

```diff
@@ -1,7 +1,30 @@
 # OverReact Analyzer Plugin
 
-> A [Dart analyzer plugin][analyzer_plugin] for OverReact
+> A [Dart analyzer plugin][analyzer_plugin] for OverReact.
 
+---
+
+* __[Goal](#goal)__
+* __[Try it in your Package](#try-it-in-your-package)__
+* __[Repo Structure](#repo-structure)__
+* __[Local Development](#local-development)__
+    * [Setup](#setup)
+    * [Development Cycle](#development-cycle)
+    * [Design Principles & Coding Strategies](#design-principles--coding-strategies)
+    * [Debugging](#debugging-the-plugin)
+* __[Feature Ideas & Inspiration](#feature-ideas--inspiration)__
+* __[Limitations](#limitations)__
+
+## Goal
+
+We want to improve the developer experience for Workiva developers writing React in Dart (and bring it closer to the React JS dev experience).
+
+This plugin has the potential to help achieve that goal by:
+* Providing static analysis of common pitfalls, with useful error messages and automatic fix suggestions.
+* Making the fluent interface syntax easier to write and deal with.
+* Helping automate common changes that can't be achieved with snippets alone (AKA "assists").
+* Encouraging best practices via lints.
+* (Potentially) Making it easier to navigate component nesting and the flow of props.
```

Comment:
This is more of a LINK marketing pitch more than anything haha. Can we remove this an then follow up later with a better-crafted description?